classlib/src/main/java/org/teavm/classlib/java/util/TMap.java

```diff
@@ -15,8 +15,11 @@
  */
 package org.teavm.classlib.java.util;
 
+import static org.teavm.classlib.java.util.TObjects.requireNonNull;
+import java.util.Objects;
 import java.util.function.BiFunction;
 import java.util.function.Function;
+import org.teavm.classlib.java.util.function.TBiConsumer;
```

Comment:
Existing code imports from `java.util.function`. Why do you import from `org.teavm.classlib.java.util.function`?

Reply:
Done.

Is there a good rule on when to use the regular java.* classes and when to use the T-classes in org.teavm.classlib.* ?
I've seen different versions of the same class used in different places, e.g.:
`static class ArrayAsList<T> extends TAbstractList<T> implements RandomAccess` in TArray vs. `public class TArrayList<E> extends TAbstractList<E> implements TCloneable, TSerializable, TRandomAccess`.
Why does one of the classes implement `TRandomAcces` and the other `RandomAccess`?

Reply:
There's a rule to always use java package when possible. Reference to `T` class is a quite old approach and I don't use it anymore.